instrumentation/jdbc/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/jdbc/test/JdbcInstrumentationTest.java

```diff
@@ -0,0 +1,1365 @@
+        Arguments.of(
+            "h2",
+            cpDatasources.get("c3p0").get("h2").getConnection(),
+            null,
+            "SELECT 3",
+            "SELECT ?",
+            "SELECT " + dbNameLower,
+            "h2:mem:",
+            null),
+        Arguments.of(
+            "derby",
+            cpDatasources.get("c3p0").get("derby").getConnection(),
+            "APP",
+            "SELECT 3 FROM SYSIBM.SYSDUMMY1",
+            "SELECT ? FROM SYSIBM.SYSDUMMY1",
+            "SELECT SYSIBM.SYSDUMMY1",
+            "derby:memory:",
+            "SYSIBM.SYSDUMMY1"),
+        Arguments.of(
+            "hsqldb",
+            cpDatasources.get("c3p0").get("hsqldb").getConnection(),
+            "SA",
+            "SELECT 3 FROM INFORMATION_SCHEMA.SYSTEM_USERS",
+            "SELECT ? FROM INFORMATION_SCHEMA.SYSTEM_USERS",
+            "SELECT INFORMATION_SCHEMA.SYSTEM_USERS",
+            "hsqldb:mem:",
+            "INFORMATION_SCHEMA.SYSTEM_USERS"));
+  }
+
+  @DisplayName(
+      "basic statement with #connection.getClass().getCanonicalName() on #system generates spans")
+  @ParameterizedTest
+  @MethodSource("basicStatementStream")
+  public void testBasicStatement(
+      String system,
+      Connection connection,
+      String username,
+      String query,
+      String sanitizedQuery,
+      String spanName,
+      String url,
+      String table)
+      throws SQLException {
+    Statement statement = connection.createStatement();
+    ResultSet resultSet = testing.runWithSpan("parent", () -> statement.executeQuery(query));
+
+    resultSet.next();
+    assertThat(resultSet.getInt(1)).isEqualTo(3);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasKind(SpanKind.INTERNAL).hasNoParent(),
+                span ->
+                    span.hasName(spanName)
+                        .hasKind(SpanKind.CLIENT)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(DbIncubatingAttributes.DB_SYSTEM, system),
+                            equalTo(DbIncubatingAttributes.DB_NAME, dbNameLower),
+                            satisfies(
+                                DbIncubatingAttributes.DB_USER,
+                                val ->
+                                    val.satisfiesAnyOf(
+                                        v -> assertThat(v).isEqualTo(username),
+                                        v -> assertThat(v).isNull())),
+                            equalTo(getDbConnectionStringKey(), url),
+                            equalTo(DbIncubatingAttributes.DB_STATEMENT, sanitizedQuery),
+                            equalTo(DbIncubatingAttributes.DB_OPERATION, "SELECT"),
+                            equalTo(DbIncubatingAttributes.DB_SQL_TABLE, table))));
+    statement.close();
+    connection.close();
```

Comment:
other tests use
```
    cleanup.deferCleanup(statement);
    cleanup.deferCleanup(connection);
```

Reply:
That is fine.
To me it feels a bit odd that there is a bunch of tests where connection is passes as parameter and still use `cleanup.deferCleanup(connection);` and there is one test that does not do it. Perhaps remove it from other tests also where it is not neede?

Reply:
done